app/api-v1/controllers/Recipe/__tests__/transactions.test.js

```diff
@@ -155,47 +150,108 @@ describe('recipe controller', () => {
 
     describe('happy path', () => {
       beforeEach(async () => {
-        nock('http://localhost:3001').post('/v3/run-process').reply(200, [20])
-        whereRecipeStub = stub().resolves([recipeExample])
+        nock(dscpApiUrl)
+          .post('/v3/run-process', function (body) {
+            runProcessBody = body
+            return true
+          })
+          .reply(200, [20])
+        stubs.getRecipe.restore()
+        stubs.insertTransaction.restore()
+        stubs.getRecipe = stub(db, 'getRecipe').resolves([recipeExample])
+        stubs.insertTransaction = stub(db, 'insertRecipeTransaction').resolves({ id: 'transaction-uuid' })
         response = await submitTransaction(postPayload)
       })
 
-      it('validates req params', () => {})
+      afterEach(() => {
+        stubs.getRecipe.restore()
+        nock.cleanAll()
+      })
 
       it('checks if recipe is in local db', () => {
-        expect(whereRecipeStub.getCall(0).args[0]).to.deep.equal({ id: 'recipe-id' })
+        expect(stubs.getRecipe.getCall(0).args[0]).to.deep.equal('recipe-id')
+      })
+
+      it('calls runProcess with correct data', () => {
+        const splitReqBody = runProcessBody.split('\n')
+        const dataHeader = splitReqBody[1]
+        const { inputs, outputs } = JSON.parse(splitReqBody[3])
+        const { roles, metadata } = outputs[0]
+
+        expect(dataHeader).to.equal('Content-Disposition: form-data; name="request"\r')
+        expect(inputs).to.deep.equal([])
+        expect(roles).to.deep.contain({
+          Owner: '5FHneW46xGXgs5mUiveU4sbTyGBzmstUspZC92UhjJM694ty',
+        })
+        expect(metadata).to.deep.contain({
+          externalId: { type: 'LITERAL', value: 'TEST-externalId' },
+          name: { type: 'LITERAL', value: 'TEST-name' },
+          material: { type: 'LITERAL', value: 'TEST-material' },
+          alloy: { type: 'LITERAL', value: 'TEST-alloy' },
+          requiredCerts: { type: 'FILE', value: 'required_certs.json' },
+          type: { type: 'LITERAL', value: 'RECIPE' },
+          image: { type: 'FILE', value: 'foo.jpg' },
+          transactionId: { type: 'LITERAL', value: 'transaction-uuid' },
+        })
       })
 
-      it('inserts new transaction to local db', async () => {
-        expect(insertTransactionStub.getCall(0).args).to.be.deep.equal([
-          {
-            recipe_id: 'recipe-id',
-            status: 'Submitted',
-            type: 'Creation',
-          },
-        ])
+      it('calls runProcess with certificate data', async () => {
+        const splitReqBody = runProcessBody.split('\n')
+        const certificatesHeader1 = splitReqBody[5]
+        const certificatesHeader2 = splitReqBody[6]
+        const certificatesRequestBody = splitReqBody[8]
+
+        expect(certificatesHeader1).to.equal(
+          'Content-Disposition: form-data; name="file"; filename="required_certs.json"\r'
+        )
+        expect(certificatesHeader2).to.equal('Content-Type: application/json\r')
```

Comment:
could we use https://github.com/nock/nock#request-headers-matching instead of `expect`?

Reply:
well, yes we could and probly should. In this instance I wanted to confirm request body. Due to a few reason, main would be that this is fire and forget.

Also nock matchHeaders, would basically check if headers do match, then it will nock the endpoint otherwise it won't mock it out (if headers dont match). So in this instance we are not asserting anything.

Reply:
We could, but my instinct is this might be preferable. Having a unified approach to assertions is helpful